routers/repo/view.go

```diff
@@ -119,6 +120,21 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 	ctx.Data["LatestCommit"] = latestCommit
 	ctx.Data["LatestCommitVerification"] = models.ParseCommitWithSignature(latestCommit)
 	ctx.Data["LatestCommitUser"] = models.ValidateCommitWithEmail(latestCommit)
+	commit := models.SignCommitWithStatuses{
+		State:    "",
+		Statuses: make([]*models.CommitStatus, 0),
+	}
+	commit.Statuses, err = models.GetLatestCommitStatus(ctx.Repo.Repository, ctx.Repo.Commit.ID.String(), 0)
```

Comment:
Why not get the CommitStatus for the latest SHA? (As-is now you're introducting race-conditions)

Reply:
This is duplicated code from ParseCommitsWithStatus function.

Reply:
Since `ctx.Repo.Commit` has parsed and not used ParseCommitsWithStatus, I will simple handle here and maybe another PR to do further improvement.